src/services/User.service.js

```diff
@@ -9,13 +9,30 @@ class UserServices {
    * @param {string} email - email of the user
    * @returns {object} user data from database
    */
-  static async findUserByEmail(email) {
+  async findUserByEmail(email) {
     const user = await db.user.findOne({
       where: { email }
     });
     if (!user) return null;
     return user;
   }
-}
 
-export default UserServices;
+  /**
+   * setUserRole - used to update a user's role by the email provided
+   * @param {object} req - request object posted to route
+   * @returns {number} number of rows updated
+   */
+  async setUserRole(req) {
+    try {
+      await db.user.update(
+        {
+          role: req.body.role
+        },
+        { where: { email: req.body.email } }
+      );
+    } catch (error) {
+      return error;
+    }
```

Comment:
Use the custom ErrorHandler:

```javascript
try {
    const user = await db.user.update(
             { role: req.body.role },
             { where: { email: req.body.email } 
    });
    return user; 
} catch (error) {
    throw new ErrorHandler(500, "an error occurred while setting role");
}
```
You should also return the updated `user` so as to prevent extra database call in the controller. It's generally a good idea to `throw` errors in your `services` and handle them in your controllers.

Reply:
I have updated the method to return an object on update and included the error handler as well.